net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+set host                host
+set domain              domain
+set tld                 tld
+set fullhost            ${host}.${domain}.${tld}
+set domaintld           ${domain}.${tld}
+set tls_certificate_name \
+                        ${fullhost}
+
+startupitem.create     yes
+
+startupitems \
+    name               calendarserver \
+    executable \
+        ${calendarserverdir}/Library/CalendarServer/${ccsname}/bin/caldavd
+
+# use defaults, edit launch daemon plists in post-activate below
+# executable ${prefix}/sbin/nginx -g "daemon off;" -p ${calendarserverdir}/Library/CalendarServer/logs -c ${calendarserverdir}/Library/CalendarServer/etc/nginx.conf
+# pidfile auto ${calendarserverdir}/Library/CalendarServer/logs/nginx.pid
+startupitems-append \
+    name               calendarserver_proxy \
+    executable         ${prefix}/sbin/nginx \
+    pidfile            auto
+
+post-activate {
+    # personal data outside of ${prefix}
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer] } {
+        xinstall -m 0755 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer
+    }
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer/Data] } {
+        xinstall -m 0700 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer/Data
+    }
+
+    # delete previous install (see destroot notes above)
+    foreach f_or_d {
+        bin
+        environment.sh
+        lib
+        roots
+        virtualenv
+        } {
+        if { [file exists \
+                  ${calendarserverdir}/Library/CalendarServer/${ccsname}/${f_or_d}]
+            } {
+            delete \
+                ${calendarserverdir}/Library/CalendarServer/${ccsname}/${f_or_d}
+        }
+    }
+
+    # macOS-native Python needs the pip and virtualenv modules
+    if { ![file executable /usr/local/bin/pip] } {
+        system "/usr/bin/easy_install pip"
+    }
+
+    # package/install calendarserver binaries
+    system -W \
+        ${prefix}/src/${ccsname}/${distname} \
+        "sh <<PACKAGE_CALENDARSERVER
+            # Don't use SecureTransport, MacPorts Python binaries or libraries
+            export USE_OPENSSL=1
+            export PATH='/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:$env(PATH)'
+            export C_INCLUDE_PATH='/usr/include:/usr/local/include:${prefix}/include'
+            export LD_LIBRARY_PATH='/usr/lib:/usr/local/lib:${prefix}/lib:${prefix}/lib/openssl-1.0'
+            export CPPFLAGS='-I/usr/include -I/usr/local/include -I${prefix}/include'
+            export LDFLAGS='-L/usr/lib -L/usr/local/lib -L${prefix}/lib -L${prefix}/lib/openssl-1.0'
+            export DYLD_LIBRARY_PATH='/usr/lib:/usr/local/lib:${prefix}/lib:${prefix}/lib/openssl-1.0'
+            ./bin/package \
+                -F ${calendarserverdir}/Library/CalendarServer/${ccsname}
+PACKAGE_CALENDARSERVER
```

Comment:
Why cannot this be done during the build phase?

Reply:
See above, https://github.com/macports/macports-ports/pull/4978#discussion_r311932707

Reply:
Per the [package author](https://github.com/apple/ccs-calendarserver/issues/540#issuecomment-519596004) @dreness, I now do `./bin/develop` in the `build` phase.

But because of the `DESTDIR` and `PREFIX` issues, we still must do `./bin/package` in `post-activate`.